lib/igniter/code/common.ex

```diff
@@ -226,20 +230,29 @@ defmodule Igniter.Code.Common do
   \"\"\"
   ```
   """
-  @spec add_code(Zipper.t(), String.t() | Macro.t(), :after | :before) :: Zipper.t()
-  def add_code(zipper, new_code, placement \\ :after)
+  @spec add_code(Zipper.t(), String.t() | Macro.t(), add_code_opts() | atom()) :: Zipper.t()
```

Comment:
A few thoughts on this spec:

- While we'll *technically* handle the atom-3rd-argument case, it'll be deprecated, so I'm fine with Dialyzer failing and only including the list of opts in the spec.
- IIRC, the best practice is to specify kw option types as `[opt()]` as opposed to `opts()`, so we could change this to `[add_code_opt()]`, but...
- Since the options aren't shared, IMO it's nicer to co-locate them, which makes them visible in hover docs in your editor, for example:

```suggestion
  @spec add_code(Zipper.t(), String.t() | Macro.t(), [opt]) :: Zipper.t()
        when opt: {:placement, :after | :before} | {:expand_env?, boolean()}
```

Reply:
Ah yeah I wasn't quite sure on best practice here - fixed in [`9f79055` (#151)](https://github.com/ash-project/igniter/pull/151/commits/9f79055b35b474f81670652481cb791ee97768e3)!